File: deck.go
package main

import "fmt"

type deck struct {
	col1 []card
	col2 []card
	col3 []card
}

func (d deck) getColumn(index int) []card {
	if index == 0 {
		return d.col1
	}
	if index == 1 {
		return d.col2
	}
	if index == 2 {
		return d.col3
	}

	return []card{}
}

func (d deck) String() string {
	const separator = "---------------" + "\n"

	allStrings := separator
	allStrings += " c1 |  c2 |  c3" + "\n"
	allStrings += separator

	for i := 1; i < len(d.col1); i++ {
		card1 := d.col1[i]
		card2 := d.col2[i]
		card3 := d.col3[i]

		allStrings += fmt.Sprintf("%v | %v | %v", card1, card2, card3)

		allStrings += "\n"
	}

	allStrings += separator

	return allStrings
}

File: deck.py
from typing import List
from card import Card

class Deck:

    col1: List[Card]
    col2: List[Card]
    col3: List[Card]

    def __init__(self, col1: List[Card], col2: List[Card], col3: List[Card]):
        self.col1 = col1
        self.col2 = col2
        self.col3 = col3

    def __str__(self):
        output = "---------------\n"
        output = output + " c1 |  c2 |  c3 \n"
        output = output + "===============\n"

        for i in range(len(self.col1)):
            card1 = self.col1[i]
            card2 = self.col2[i]
            card3 = self.col3[i]
            output = output + f"{card1} | {card2} | {card3}\n"

        output = output + "---------------\n"
        return output
